assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -108,40 +97,37 @@ function useUploader() {
         throw createError('SizeError', file.name, message);
       }
 
-      const additionalData = {
-        post: storyId,
-        media_source: 'editor',
-        ..._additionalData,
-      };
+      // The file is too large for the site anyway, abort.
+      if (!isFileSizeWithinLimits(file)) {
```

Comment:
I think we need to add a check for `&& !canTranscodeFile) {` here. I was unable to upload a 164MB mp4.

Reply:
Not so sure about that.

In your screenshot, your upload limit is 100MB and the file is 164MB.

Yes, the file could be easily transcoded, _but_ there is no guarantee that transcoding will make the file < 100 MB (we simply don't know). So it makes sense to just abort because of the upload limit.

I'd think this is also what happens on `main`

Reply:
Confirmed that this is the same behavior as on `main`:

https://github.com/google/web-stories-wp/blob/7ea172b9292455e1e86ee2b0ac71a23e5cddfb49/assets/src/edit-story/app/uploader/useUploader.js#L97-L141

I would not want to change this in this PR.

Reply:
As we have seen, transcoding brings the file size down by a lot. Can we take the file that is too long, transcode it and then try and upload it? If a 162mb becomes at 16mb file, then there is no problem. This would need a new error, of "We tried to transcode your file but it was too big to upload after it was transcoded". This discussion may need to be moved out another ticket.

Reply:
Yeah I can create a new ticket for that. Would be good to have UX feedback on it.

My worry there is just that we'll spend multiple minutes trying to transcode & upload file only to tell the user that it won't work.

But with the right messaging it might work.

Reply:
> My worry there is just that we'll spend multiple minutes trying to transcode & upload file only to tell the user that it won't work.

Also my worry as well, it might be a waste. But we do cap about at 2GB, so if you try and upload star wars, that limitation should kick in. Marking this as resolved for now, let's chat on the new ticket.

Reply:
Ticket created: #6580